agent/proxycfg/snapshot.go

```diff
@@ -451,7 +455,8 @@ func (s *ConfigSnapshot) Valid() bool {
 		}
 		return s.Roots != nil &&
 			s.ConnectProxy.Leaf != nil &&
-			s.ConnectProxy.IntentionsSet
+			s.ConnectProxy.IntentionsSet &&
+			s.ConnectProxy.TLSConfigSet
```

Comment:
Is the TLS config a required field before sending back resources so that we don't reconfigure the transport socket of a listener unnecessarily?

Reply:
Yes, so that the first time you try to do TLS you follow the user's intent and don't introduce a racy downgrade window.